File: rpc/2020/rpc01/local2019Data/solutions/divide/divide.py
#!/usr/bin/env python

# Jim Geist
# 8/27/2019
# Solution to 2019 UCF Local Contest Problem: Divide
#

import sys

# read all the numbers
data = [int(x) for x in ' '.join(sys.stdin.readlines()).split()]

# get the 'n' and 'd' parameters
n = data[0]
d = data[1]
data = data[2:n+2]

# compute the total problems solved
total_solved = sum(data)

# compute the award for each problem
award = d / total_solved

# write out each student's award
for solved in data:
    print(award * solved)

File: rpc/2020/rpc01/local2019Data/solutions/divide/divide.cpp
// Jim Geist
// 8/27/2019
// Solution to 2019 UCF Local Contest Problem: Divide
//

#include <iostream>

int main()
{
    int n, d;

    // read n and d
    std::cin >> n >> d;

    int solved[30];

    // read how many problems each student solved, and also compute
    // the total number of problems solved
    //
    int total_problems = 0;
    for (int i = 0; i < n; i++) {
        std::cin >> solved[i];
        total_problems += solved[i];
    }

    // how much money to award for each problem
    //
    int award = d / total_problems;

    // now we can give out money!
    //
    for (int i = 0; i < n; i++) {
        std::cout << award * solved[i] << std::endl;
    }
}